refmt: init at 1.16.0

deemp opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

emilylange reviewed (changes requested):
Please edit your PR to target the `master` branch instead of `nixpkgs-unstable`.
Thank you :)

Please also read the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/da072c0d594456cd309f79785b4a97408e3562ac/CONTRIBUTING.md) (that tick box in your PR description). Specifically "Submitting changes" (as your commit message needs to be `refmt: init at 1.16.0`) and the `meta.description` section.

deemp commented:
@emilylange, thank you!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2315

jsoo1 reviewed (approved):
LGTM

emilylange reviewed (changes requested):
Oh wait.
Can you please split the change to `maintainers/maintainer-list.nix` into its own separate commit called `maintainers: add deemp`?

So two commits total:
1. `maintainers: add deemp`
1. `refmt: init at 1.16.0`

See https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

deemp committed:
maintainers: add deemp

deemp commented:
@emilylange, done!

emilylange commented on pkgs/development/tools/refmt/default.nix:
```diff
@@ -0,0 +1,22 @@
+  meta = with lib; {
+    description = "Reformat HCL <-> JSON <-> YAML";
+    homepage = "https://github.com/rjeczalik/refmt";
+    license = licenses.agpl3;
```
One more thing, sorry^^

```suggestion
    license = licenses.agpl3Only;
```

deemp replied:
Done!

deemp committed:
refmt: init at 1.16.0

About: Reformat HCL ⇄ JSON ⇄ YAML.
Release: https://github.com/rjeczalik/refmt/releases/tag/v1.6.1

emilylange reviewed (approved):
Thank you! :)
